Add build variations + refactor breaking-changes. (#3478)
dapr/docs: daprdocs/content/en/operations/support/support-release-policy.md

Patch support is for supported versions (current and previous).

## Build variations

> The Dapr's sidecar image is published to GitHub Container Registry and Docker Registry. The default image contains all components. Since version 1.11, Dapr also offers a variation of the sidecar image, containing only stable components.

Reviewer: ```suggestion
The Dapr's sidecar image is published to both [GitHub Container Registry](https://github.com/dapr/dapr/pkgs/container/daprd) and [Docker Registry](https://hub.docker.com/r/daprio/daprd/tags). The default image contains all components. From version 1.11, Dapr also offers a variation of the sidecar image, containing only stable components.
```

> * Default sidecar images: `daprio/daprd:<version>` or `ghcr.io/dapr/daprd:<version>`

Reviewer: ```suggestion
* Default sidecar images: `daprio/daprd:<version>` or `ghcr.io/dapr/daprd:<version>`. For example, `daprio/daprd:1.11.0-linux-amd64`
```

Reviewer: I may have this wrong above. My question is where does the tag `daprio/daprd:latest` come in here, since I presume this is the default image? Should we say something here?

Author: Platform is not required, since the -stable tag is multi-arch and K8s downloads the right arch automatically.

> * Sidecar images for stable components: `daprio/daprd:<version>-stable` or `ghcr.io/dapr/daprd:<version>-stable`

Reviewer: ```suggestion
* Sidecar images for stable components: `daprio/daprd:<version>-stable-<platform>` or `ghcr.io/dapr/daprd:<version>-stable-<platform>`. For example, `daprio/daprd:1.11.0-stable-linux-amd64`
```

Author: Same, platform is optional.

Author: See updated version with example.

> * Default sidecar images: `daprio/daprd:<version>` or `ghcr.io/dapr/daprd:<version>` (Example: `ghcr.io/dapr/daprd:1.11.0`)

Reviewer: ```suggestion
* Default sidecar images: `daprio/daprd:<version>` or `ghcr.io/dapr/daprd:<version>` (for example `ghcr.io/dapr/daprd:1.11.0`)
```

> * Sidecar images for stable components: `daprio/daprd:<version>-stable` or `ghcr.io/dapr/daprd:<version>-stable` (Example: `ghcr.io/dapr/daprd:1.11.0-stable`)

Reviewer: ```suggestion
* Sidecar images for stable components: `daprio/daprd:<version>-stable` or `ghcr.io/dapr/daprd:<version>-stable` (for example `ghcr.io/dapr/daprd:1.11.0-stable`)
```

> On Kubernetes, the sidecar image can be overwritten at the Deployment resource of the application via the `dapr.io/sidecar-image` annotation. See more about [Dapr's arguments and annotations]({{<ref "arguments-annotations-overview.md" >}})

Reviewer: ```suggestion
On Kubernetes, the sidecar image can be overwritten for the application Deployment resource with `dapr.io/sidecar-image` annotation. See more about [Dapr's arguments and annotations]({{<ref "arguments-annotations-overview.md" >}}). The default 'daprio/daprd:latest' image is used if not specified. 
```

Reviewer: I do suggest that we also update https://docs.dapr.io/reference/arguments-annotations-overview/ into this PR for the dapr.io/sidecar-image annotations and have a link back to this topic. Like this:

"Dapr sidecar image. Default is daprio/daprd:latest. The Dapr sidecar will use this image instead of the latest default image. Use this when building your own custom image of Dapr and or [using the alternative stable Dapr image]({{<ref "support-release-policy.md/#build-variations" >}}) .

Author: Done.

> On Kubernetes, the sidecar image can be overwritten at the Deployment resource of the application via the `dapr.io/sidecar-image` annotation. See more about [Dapr's arguments and annotations]({{<ref "arguments-annotations-overview.md" >}})

Reviewer: ```suggestion
On Kubernetes, the sidecar image can be overwritten for the application Deployment resource with the `dapr.io/sidecar-image` annotation. See more about [Dapr's arguments and annotations]({{<ref "arguments-annotations-overview.md" >}}). The default 'daprio/daprd:latest' image is used if not specified.
```
